.github/workflows/ci.yml

```
@@ -0,0 +1,117 @@
+name: CI Checks
+
+on:
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+
+jobs:
+  build-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: CMake
+        run: |
+          cmake -S . -B build/ \
+          -G "Unix Makefiles" \
+          -DBUILD_TESTS=1 \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -Werror' \
+          -DAWS_IOT_ENDPOINT="aws-iot-endpoint" \
+          -DBROKER_ENDPOINT="broker-endpoint" \
+          -DCLIENT_CERT_PATH="cert/path" \
+          -DROOT_CA_CERT_PATH="cert/path" \
+          -DCLIENT_PRIVATE_KEY_PATH="key/path"
+      - name: Build Demos
+        run: make -C build/ help | grep demo | tr -d '. ' | xargs make -C build/
+      - name: Build System Tests
+        run: make -C build/ help | grep system | tr -d '. ' | xargs make -C build/
+  unittest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Build
+        run: |
+          sudo apt-get install -y lcov
```

Is the `-y  flag required?
```suggestion
          sudo apt-get install  lcov
```

---

It might be in another environment.